Sources/Publish/API/AnyDateFormatter.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AnyDateFormatter.swift
+//  
+//
+//  Created by Dustin Pfannenstiel on 3/10/20.
+//  Copyright (c) Dustin Pfannenstiel, 2020
+//  MIT license, see LICENSE file for details
+//
+
+import Foundation
+
+/// A common interface to abstract common methods and properties present
+/// in all date formatter types.
+public protocol AnyDateFormatter: Formatter {
+    /// The time zone for the receiver.
+    var timeZone: TimeZone! { get set }
+    /// The date format string used by the receiver.
+    var dateFormat: String! { get }
```

Comment:
This API is not used within Publish, so including it feels a bit like premature generalization, and makes the `AnyDateFormatter` protocol more complex than it has to be.

Reply:
This is actually used in `MarkdownMetadataDecoder` [here](https://github.com/JohnSundell/Publish/blob/6a2003f5ed2019cb2f566339f81f148726ce1b67/Sources/Publish/Internal/MarkdownMetadataDecoder.swift#L627).